Jenkins test failure in test15_storageStatistics

Fails on Plone 5 AT tests. See http://jenkins.plone.org/job/plone-5.0-python-2.7-at/3376/
The failure started after an update to mockup, but it did not go away when I reverted this, so I reverted my revert. BTW, the original mockup commits were not from me.
Also nice: I do not get the test failure locally.

The failure for good measure:

```
Failure in test test15_storageStatistics (Products.CMFEditions.tests.test_ZVCStorageTool.TestZVCStorageTool)
Traceback (most recent call last):
  File "/usr/lib/python2.7/unittest/case.py", line 331, in run
    testMethod()
  File "/home/jenkins/.buildout/eggs/Products.CMFEditions-2.2.16-py2.7.egg/Products/CMFEditions/tests/test_ZVCStorageTool.py", line 555, in test15_storageStatistics
    self.assertEqual(expected, got)
  File "/usr/lib/python2.7/unittest/case.py", line 515, in assertEqual
    assertion_func(first, second, msg=msg)
  File "/usr/lib/python2.7/unittest/case.py", line 837, in assertDictEqual
    self.fail(self._formatMessage(msg, standardMsg))
  File "/usr/lib/python2.7/unittest/case.py", line 412, in fail
    raise self.failureException(msg)
AssertionError: {'deleted': [], 'summaries': {'totalHistories': 4, 'deletedVersions': 0, 'existi [truncated]... != {'deleted': [], 'summaries': {'totalHistories': 4, 'deletedVersions': 0, 'existi [truncated]...
  {'deleted': [],
   'existing': [{'history_id': 1,
                 'length': 4,
                 'path': '/obj',
                 'portal_type': 'Dummy',
-                'size': 1718,
?                          ^

+                'size': 1738,
?                          ^

                 'sizeState': 'approximate',
                 'url': 'http://nohost/plone/obj'},
                {'history_id': 2,
                 'length': 1,
                 'path': '/tomorrow',
                 'portal_type': 'Dummy',
-                'size': 555,
?                         ^^

+                'size': 560,
?                         ^^

                 'sizeState': 'approximate',
                 'url': 'http://nohost/plone/tomorrow'},
                {'history_id': 3,
                 'length': 1,
                 'path': '/yesterday',
                 'portal_type': 'Dummy',
-                'size': 557,
?                         ^^

+                'size': 562,
?                         ^^

                 'sizeState': 'approximate',
                 'url': 'http://nohost/plone/yesterday'},
                {'history_id': 4,
                 'length': 1,
                 'path': '/public',
                 'portal_type': 'Dummy',
-                'size': 557,
?                         ^^

+                'size': 562,
?                         ^^

                 'sizeState': 'approximate',
                 'url': 'http://nohost/plone/public'}],
   'summaries': {'deletedAverage': 'n/a',
                 'deletedHistories': 0,
                 'deletedVersions': 0,
                 'existingAverage': '1.8',
                 'existingHistories': 4,
                 'existingVersions': 7,
                 'time': '0.00',
                 'totalAverage': '1.8',
                 'totalHistories': 4,
                 'totalVersions': 7}}

  Ran 152 tests with 1 failures, 0 errors and 0 skipped in 47.890 seconds.
```

--- a/CHANGES.rst
+++ b/CHANGES.rst
@@ -10,7 +10,8 @@ New:
 
 Fixes:
 
-- *add item here*
+- make storage statistics test more robust - see https://github.com/plone/Products.CMFEditions/issues/31
+  [tschorr]
 
 
 2.2.16 (2015-09-27)

--- a/Products/CMFEditions/tests/test_ZVCStorageTool.py
+++ b/Products/CMFEditions/tests/test_ZVCStorageTool.py
@@ -526,7 +526,6 @@ class TestZVCStorageTool(CMFEditionsBaseTestCase):
                             'path': '/obj', 
                             'sizeState': 'approximate', 
                             'portal_type': 'Dummy', 
-                            'size': 1718
                         }, {
                             'url': 'http://nohost/plone/tomorrow', 
                             'history_id': 2, 
@@ -534,7 +533,6 @@ class TestZVCStorageTool(CMFEditionsBaseTestCase):
                             'path': '/tomorrow', 
                             'sizeState': 'approximate', 
                             'portal_type': 'Dummy', 
-                            'size': 555
                         }, {
                             'url': 'http://nohost/plone/yesterday', 
                             'history_id': 3, 
@@ -542,7 +540,6 @@ class TestZVCStorageTool(CMFEditionsBaseTestCase):
                             'path': '/yesterday', 
                             'sizeState': 'approximate', 
                             'portal_type': 'Dummy', 
-                            'size': 557
                         }, {
                             'url': 'http://nohost/plone/public', 
                             'history_id': 4, 
@@ -550,9 +547,18 @@ class TestZVCStorageTool(CMFEditionsBaseTestCase):
                             'path': '/public', 
                             'sizeState': 'approximate', 
                             'portal_type': 'Dummy', 
-                            'size': 557
                         }]}
-        self.assertEqual(expected, got)
+        self.assertEqual(expected['deleted'], got['deleted'])
+        self.assertEqual(expected['summaries'], got['summaries'])
+        self.assertEqual(len(expected['existing']), len(got['existing']))
+        for idx in range(len(expected['existing'])):
+            e = expected['existing'][idx]
+            g = got['existing'][idx]
+            for k, v in e.items():
+                self.assertEqual(g[k], v)
+            # The actual size is not important and we want robust tests,
+            # s. https://github.com/plone/Products.CMFEditions/issues/31
+            self.failUnless(g['size'] > 0)
 
 
 class TestMemoryStorage(TestZVCStorageTool):

